ros_cross_compile/ros_cross_compile.py

```diff
@@ -174,7 +184,15 @@ def cross_compile_pipeline(
 def main():
     """Start the cross-compilation workflow."""
     args = parse_args(sys.argv[1:])
-    cross_compile_pipeline(args)
+    data_collector = DataCollector()
+
+    with DataTimer() as end_end_timer:
+        cross_compile_pipeline(args, data_collector)
+
+    end_end_time = Datum('End_To_End-time', end_end_timer.elapsed,
```

Comment:
This still requires too much specialized knowledge on the part of the user. Let's figure out the implementation such that the entire block looks like

```
data_collector = DataCollector()

with data_collector.timer("end-to-end"):
  cross_compile_pipeline(args, data_collector)

data_collector.write_file()
```